psi4/driver/qcdb/vib.py

```diff
@@ -0,0 +1,1062 @@
+
+
+def compare_vibinfos(expected, computed, tol, label, verbose=1, forgive=None, required=None):
+    """Compare two dictionaries of VibrationAspect objects. All items in
+    `expected` must be present in `computed` and agree to `tol` (outright
+    (<1) or decimal places (>=1)). At minimum, both must contain items in
+    `required` ('omega' recc. for vibs).  Failed items in `forgive` will
+    not count against passing.
+
+    """
+    import numpy as np
+
+    np.set_printoptions(formatter={'float': '{: 0.4f}'.format})
+
+    def _success(label):
+        """Function to print a '*label*...PASSED' line to screen.
+        Used by :py:func:`util.compare_values` family when functions pass.
+        """
+        msg = '\t{0:.<66}PASSED'.format(label)
+        print(msg)
+        sys.stdout.flush()
+
+    def print_stuff(asp, same, ref, val, space=''):
+        if verbose >= 1:
+            print(asp, ':', same)
+        if (verbose >= 2) or (not same and verbose >= 1):
+            print('\texp:', space, ref)
+            print('\tobs:', space, val)
+        if verbose >= 1:
+            if not same:
+                try:
+                    print('\tdif:', space, val - ref)
+                except TypeError:
+                    print('\tdif: Different, inspect arrays')
+
+    if tol >= 1.0:
+        tol = 10.**-tol
+
+    if forgive is None:
+        forgive = []
+
+    summsame = []
+    if required is None:
+        checkkeys = []
+    else:
+        checkkeys = required
+    checkkeys.extend(expected.keys())
+
+    for asp in checkkeys:
+        if asp in ['q', 'w', 'x']:
+            ccnc = _phase_cols_to_max_element(computed[asp].data)
+            eenc = _phase_cols_to_max_element(expected[asp].data)
+            ccnc = _check_degen_modes(ccnc, computed['omega'].data)
+            eenc = _check_degen_modes(eenc, expected['omega'].data)
+            same = np.allclose(eenc, ccnc, atol=tol)
+            print_stuff(asp=asp, same=same, ref=eenc, val=ccnc, space='\n')
+
+        elif asp in ['gamma', 'TRV']:
+            same = all([computed[asp].data[idx] == val for idx, val in enumerate(expected[asp].data)])
+            print_stuff(asp=asp, same=same, ref=expected[asp].data, val=computed[asp].data)
+
+        elif isinstance(expected[asp].data, float):
+            same = abs(expected[asp].data - computed[asp].data) < tol
+            print_stuff(asp=asp, same=same, ref=expected[asp].data, val=computed[asp].data)
+        else:
+            same = np.allclose(expected[asp].data, computed[asp].data, atol=tol) and \
+                   (expected[asp].data.shape == computed[asp].data.shape)
+            print_stuff(asp=asp, same=same, ref=expected[asp].data, val=computed[asp].data)
+
+        if asp not in forgive:
+            summsame.append(same)
+
+    passed = all(summsame)
+    if passed:
+        _success(label)
+    return passed
+
+
+def hessian_symmetrize(hess, mol):
```

Comment:
Yay!  Thanks for figuring out how to do this properly, so we can have a ceremonial burning of my horribly failed attempt.